### PR TITLE
[App Service] Add support for vnet route all setting

### PR DESCRIPTION
**Pull request by madsd**
**Description**
Adding support for vnet route all setting in config set
Defaulting to vnet_route_all in webapp vnet-integration add
Add support for skipping delegation in webapp vnet-integration add (for scenarios with limited permission on subnet)
Adding support for cross-subscription vnet-integration
Fixing minor syntax error in help file

**Testing Guide**
az webapp config set -g <group> -n <webapp> --vnet-route-all-enabled
az webapp config set -g <group> -n <webapp> --vnet-route-all-enabled false

az webapp vnet-integration add -g <group> -n <webapp> --vnet <vnet-name> --subnet <subnet> [--skip-delegation-check]

**History Notes**

[AppService] az webapp config set: Add support for VNet Route All property.
[AppService] az webapp vnet-integration add: Default to VNet Route All. Allow cross subscription integration.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -1248,7 +1248,7 @@
         az webapp config set -g MyResourceGroup -n MyUniqueApp --always-on true
   - name: turn on "alwaysOn" through a json with content "{"alwaysOn", true}"
```
[the remark this answers is not in the document]
Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py]: Good catch - pushed an alignment with the example change.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -353,6 +353,8 @@ def load_arguments(self, _):
                        help="The startup file for linux hosted web apps, e.g. 'process.json' for Node.js web")
             c.argument('ftps_state', help="Set the Ftps state value for an app. Default value is 'AllAllowed'.",
                        arg_type=get_enum_type(FTPS_STATE_TYPES))
+            c.argument('vnet_route_all_enabled', help="configures regional VNet integration to route all traffic to the VNet.",
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:356]:
```suggestion
            c.argument('vnet_route_all_enabled', help="Configure regional VNet integration to route all traffic to the VNet.",
```

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -3584,31 +3585,48 @@ def add_vnet_integration(cmd, name, resource_group_name, vnet, subnet, slot=None
...
-    subnetObj = vnet_client.subnets.get(vnet_resource_group, vnet_name, subnet_name)
-    delegations = subnetObj.delegations
-    delegated = False
-    for d in delegations:
-        if d.service_name.lower() == "microsoft.web/serverfarms".lower():
-            delegated = True
+    if skip_delegation_check:
+        logger.warning('Skipping delegation check. Ensure that subnet is delegated to Microsoft.Web/serverFarms.'
+                       ' Missing delegation can cause "Bad Request" error.')
+    else:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3596]: `else` means None or False, right? So when the value is None, you will also check delegation, right?

**Review by Juliehzl — CHANGES_REQUESTED** [review]
Please add test for the new parameters and make CI pass

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 2 above)
Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py]: Corrected

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 3 above)
Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:3596]: Yes - delegation check should always be done, unless explicitly opting out.

**Comment by madsd** [comment]
@Juliehzl Comments addressed, new test added and failing tests have been rerun.
Thanks.

**Review by Juliehzl — APPROVED** [review]
LGTM now. @panchagnula for review.

**Review by panchagnula — APPROVED** [review]
LGTM

**Comment by madsd** [comment]
@zhoxing-ms Hey Xing, are you to approve and merge this?

**Comment by madsd** [comment]
@yonzhan Is Xing OOF? I would really like to get this into S189. Or can @Juliehzl merge this.

**Comment by Juliehzl** [comment]
HI @madsd, please resolve all conflicts, then I could merge the PR

**Comment by madsd** [comment]
@Juliehzl Done

**Comment by Juliehzl** [comment]
please fix linter failure as shown in CI

FAIL - HIGH severity: missing_group_help
    Command-Group: `functionapp log` - Missing help

**Comment by madsd** [comment]
@Juliehzl hmm, this was not related to the PR, so I wonder how the code for the functionapp log was merged in the first place. Perhaps the linter rules have changes. Anyways, I added a help snippet to fix it.

**Comment by madsd** [comment]
@Juliehzl friendly nudge to merge, so it does not go stale again :-)

**Pull request merged**